### DISCUSSION TITLE
KateX can not render in my webstie

### DISCUSSION THREAD

**Discussion by koktlzz** [thuliteio/doks#382] [Q&A] [answered]
I add  KaTeX to my [website](https://koktlzz.github.io/infra/csapp/%E4%BF%A1%E6%81%AF%E7%9A%84%E8%A1%A8%E7%A4%BA%E5%92%8C%E5%A4%84%E7%90%86/)  following the steps in the [tutorial](https://getdoks.org/docs/extensions/katex/), but the formula can not render:

![20210708223052](https://cdn.jsdelivr.net/gh/koktlzz/ImgBed@master/20210708223052.png)

This is my package.json:

```json
{
  "name": "doks",
  "version": "0.1.6",
  "private": true,
  "hugo-bin": {
    "buildTags": "extended"
  },
  "browserslist": [
    "defaults"
  ],
  "scripts": {
    "init": "rimraf .git && git init -b main",
    "create": "hugo new",
    "prestart": "npm run clean",
    "start": "hugo server --disableFastRender",
    "prebuild": "npm run clean",
    "build": "hugo --gc --minify && npm run build:functions",
    "build:functions": "netlify-lambda build assets/lambda",
    "build:preview": "npm run build -D -F",
    "clean": "rimraf public resources functions",
    "lint": "npm run -s lint:scripts && npm run -s lint:styles && npm run -s lint:markdown",
    "lint:scripts": "eslint assets/js assets/lambda config",
    "lint:styles": "stylelint \"assets/scss/**/*.{css,sass,scss,sss,less}\"",
    "lint:markdown": "markdownlint *.md content/**/*.md",
    "release": "standard-version",
    "release:dry-run": "standard-version --dry-run",
    "copy:katex-fonts": "shx cp ./node_modules/katex/dist/fonts/* ./static/fonts/",
    "server": "hugo server",
    "test": "npm run -s lint",
    "env": "env",
    "precheck": "npm version",
    "check": "hugo version"
  },
  "devDependencies": {
    "@babel/cli": "^7.14.5",
    "@babel/core": "^7.14.6",
    "@babel/preset-env": "^7.14.7",
    "@fullhuman/postcss-purgecss": "^4.0.3",
    "autoprefixer": "^10.2.6",
    "bootstrap": "^4.6.0",
    "clipboard": "^2.0.8",
    "eslint": "^7.30.0",
    "flexsearch": "^0.6",
    "hugo-bin": "^0.69",
    "instant.page": "^5.1",
    "lazysizes": "^5.3.2",
    "markdownlint-cli": "^0.27",
    "netlify-lambda": "^2.0.11",
    "postcss": "^8.3.5",
    "postcss-cli": "^8.3",
    "purgecss-whitelister": "^2.4",
    "rimraf": "^3.0",
    "standard-version": "^9.3.0",
    "stylelint": "^13.13.1",
    "stylelint-config-standard": "^20.0"
  },
  "dependencies": {
    "@popperjs/core": "^2.9.2",
    "jquery": "^3.6.0",
    "katex": "^0.13.11",
    "popper.js": "^1.16.1",
    "shx": "^0.3.3"
  }
}
```

**Comment by h-enk** [comment, marked as answer]
Hello @koktlzz, you're running an older version of Doks (before KaTeX support). What I see, you'll also need to update:

- `./layouts/partials/footer/script-footer.html` — see: https://github.com/h-enk/doks/blob/master/layouts/partials/footer/script-footer.html
- `./assets/scss/app.scss` — see also: https://github.com/h-enk/doks/blob/master/assets/scss/app.scss

Maybe there's more: 

- open the latest Doks repo in e.g. Visual Studio Code
- run a "Find in Files" (Ctrl+Shift+F) for "katex"
- compare with what you have /are missing + update accordingly

That should do the trick.

**Reply by koktlzz** [reply]
Thank you for your reply! I update my doks to v0.3.3 and kateX works. I used the starter theme, so the update took me a long time :) Anyway, it works and deploys on gh-pages really faster than v0.1.

I also add highlight.js to my website following the [example](https://getdoks.org/docs/extensions/highlight-js/#example)，and i think the import statement in `app.scss` should be:

```scss
/** Import highlight.js */
@import "highlight.js/scss/base16/dracula";
```

Thanks again for your work!
